src/engine/input/key-mapping/InputEventItem.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using Godot;
+
+public class InputEventItem : Node
+{
+    [Export]
+    public NodePath ButtonPath;
+
+    internal InputActionItem AssociatedAction;
+
+    internal bool JustAdded;
+
+    private Button button;
+
+    private bool waitingForInput;
+
+    public InputEventWithModifiers AssociatedEvent { get; set; }
+
+    public void Delete()
+    {
+        AssociatedAction.AssociatedGroup.GetParent<InputGroupList>().ControlSchemeChanged();
+        AssociatedAction.Inputs.Remove(this);
+    }
+
+    public override void _Input(InputEvent @event)
+    {
+        if (InputGroupList.Instance.IsConflictDialogOpen())
+            return;
+
+        if (!waitingForInput)
+            return;
+
+        if (!(@event is InputEventMouseButton) && !(@event is InputEventKey))
+            return;
+
+        if (@event is InputEventKey iek)
+        {
+            switch (iek.Scancode)
+            {
+                case (uint)KeyList.Escape:
+                    Delete();
+                    return;
+                case (uint)KeyList.Alt:
+                case (uint)KeyList.Shift:
+                case (uint)KeyList.Control:
+                    return;
+            }
+        }
+
+        var old = AssociatedEvent;
+        AssociatedEvent = (InputEventWithModifiers)@event;
+        var conflict = InputGroupList.Instance.Conflicts(this);
```

Comment:
I think that `this` shouldn't be passed to get conflicts. Instead whatever info Conflicts wants from this should be passed in.

Reply:
this is outdated and the InputEventItem is now required in the Conflicts method.

Reply:
I still think Conflicts should be coded so that it doesn't rely on this specific InputEventItem class.